src/lib/core/theming/_theming.scss

```diff
@@ -82,3 +82,51 @@
     background: $md-dark-theme-background,
   );
 }
+
+
+// Import all the theming functionality.
+@import '../core';
+@import '../../button/button-theme';
+@import '../../button-toggle/button-toggle-theme';
+@import '../../card/card-theme';
+@import '../../checkbox/checkbox-theme';
+@import '../../dialog/dialog-theme';
+@import '../../grid-list/grid-list-theme';
+@import '../../icon/icon-theme';
+@import '../../input/input-theme';
+@import '../../list/list-theme';
+@import '../../menu/menu-theme';
+@import '../../progress-bar/progress-bar-theme';
+@import '../../progress-circle/progress-circle-theme';
+@import '../../radio/radio-theme';
+@import '../../sidenav/sidenav-theme';
+@import '../../slide-toggle/slide-toggle-theme';
+@import '../../slider/slider-theme';
+@import '../../tabs/tabs-theme';
+@import '../../toolbar/toolbar-theme';
+@import '../../tooltip/tooltip-theme';
+
+
+// Create a theme.
+@mixin angular-material-theme($theme) {
```

Comment:
As discussed: internal presubmit revealed that moving this here caused an unresolvable circular dependency. Need to move it back to it's own file. I suggest we call it `_all-component-themes.scss`